scripts/crate/index.mts

```diff
@@ -0,0 +1,54 @@
+import { cliArguments } from '../helpers/utils.mts';
+import { getClientCrateEnvironment } from './client.mts';
+import { getInterfaceCrateEnvironment } from './interface.mts';
+import { getProgramCrateEnvironment } from './program.mts';
+
+export type CrateEnvironment = {
+  getFormatArguments: (cliArguments: string[]) => string[];
+  getLintClippyArguments: (cliArguments: string[]) => string[];
+  getLintDocsArguments: (cliArguments: string[]) => string[];
+  getLintFeaturesArguments: (cliArguments: string[]) => string[];
+  getPublishArguments: (cliArguments: string[]) => string[];
+  getTestArguments: (cliArguments: string[]) => string[];
+  getWasmArguments: (cliArguments: string[]) => string[];
+  cratePath: string;
+  manifestPath: string;
+};
+
+type Crate = 'interface' | 'client' | 'program';
+
+export function getCrateEnvironmentFromArgs(): [CrateEnvironment, string[]] {
+  const crate = process.argv[2] as Crate;
+  return [getCrateEnvironment(crate), cliArguments().slice(1)];
+}
+
+export function getCrateEnvironment(crate: Crate): CrateEnvironment {
+  switch (crate) {
+    case 'client':
+      return getClientCrateEnvironment();
+    case 'interface':
+      return getInterfaceCrateEnvironment();
+    case 'program':
+      return getProgramCrateEnvironment();
+    default:
+      throw new Error(`Unknown crate: ${crate}`);
+  }
+}
```

Comment:
How much of a "script machine" do we want this whole thing to be though. There's a very hard balance to find between providing a bunch of scripts that people can adjust to their need versus offering a CLI with bells and whistles (at which point it might as well be extracted as an open source library).

Reply:
I think I'm the wrong guy to ask because I think this is starting to make more sense as a CLI/library we just reuse, because this has moved pretty far away from `create-solana-program` boilerplate scripts in my opinion.

Reply:
The goal is to eventually do the same to `create-solana-program` whenever we find the right balance. So maybe a custom CLI is on the table.

Reply:
The `crate` folder under `scripts` was removed – it is not needed anymore.